OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -314,32 +364,49 @@ bool TracerThread::OpenRingBufferForGpuTracepoint(
 // relevant events.
 // This method returns true on success, otherwise false.
 bool TracerThread::OpenGpuTracepoints(const std::vector<int32_t>& cpus) {
-  std::vector<int> gpu_tracing_fds;
+  std::vector<int> amdgpu_cs_ioctl_fds;
+  std::vector<int> amdgpu_sched_run_job_fds;
+  std::vector<int> dma_fence_signaled_fds;
   std::vector<PerfEventRingBuffer> gpu_ring_buffers;
   for (int32_t cpu : cpus) {
     if (!OpenRingBufferForGpuTracepoint("amdgpu", "amdgpu_cs_ioctl", cpu,
-                                        &gpu_tracing_fds, &gpu_ring_buffers)) {
-      CloseFileDescriptors(gpu_tracing_fds);
+                                        &amdgpu_cs_ioctl_fds,
+                                        &gpu_ring_buffers)) {
+      CloseFileDescriptors(amdgpu_cs_ioctl_fds);
       return false;
     }
     if (!OpenRingBufferForGpuTracepoint("amdgpu", "amdgpu_sched_run_job", cpu,
-                                        &gpu_tracing_fds, &gpu_ring_buffers)) {
-      CloseFileDescriptors(gpu_tracing_fds);
+                                        &amdgpu_sched_run_job_fds,
+                                        &gpu_ring_buffers)) {
+      CloseFileDescriptors(amdgpu_cs_ioctl_fds);
+      CloseFileDescriptors(amdgpu_sched_run_job_fds);
       return false;
     }
     if (!OpenRingBufferForGpuTracepoint("dma_fence", "dma_fence_signaled", cpu,
-                                        &gpu_tracing_fds, &gpu_ring_buffers)) {
-      CloseFileDescriptors(gpu_tracing_fds);
+                                        &dma_fence_signaled_fds,
+                                        &gpu_ring_buffers)) {
+      CloseFileDescriptors(amdgpu_cs_ioctl_fds);
+      CloseFileDescriptors(amdgpu_sched_run_job_fds);
+      CloseFileDescriptors(dma_fence_signaled_fds);
       return false;
     }
   }
 
   // Since all tracepoints could successfully be opened, we can now commit all
   // file descriptors and ring buffers to the TracerThread members.
-  for (int fd : gpu_tracing_fds) {
+  for (int fd : amdgpu_cs_ioctl_fds) {
+    tracing_fds_.push_back(fd);
+    amdgpu_cs_ioctl_ids_.insert(perf_event_get_id(fd));
+  }
+  for (int fd : amdgpu_sched_run_job_fds) {
+    tracing_fds_.push_back(fd);
+    amdgpu_sched_run_job_ids_.insert(perf_event_get_id(fd));
+  }
+  for (int fd : dma_fence_signaled_fds) {
     tracing_fds_.push_back(fd);
-    gpu_tracing_ids_.insert(perf_event_get_id(fd));
+    dma_fence_signaled_ids_.insert(perf_event_get_id(fd));
   }
+  // TODO: Consider redirecting all GPU events on a CPU on the same ring buffer.
```

Comment:
Isn't this done above? Or do you mean to do all three types of events on the same ring buffer (per CPU).